### PR TITLE
[easy] Refactoring to option/Result idioms

### PR DESCRIPTION
**Pull request by huitseeker**
## Motivation

Proper `Option`/`Result` patterns are easier to read, shorter, and less error-prone.

### CODE REVIEW

**Comment by dariorussi** [comment]
@huitseeker could we please start adding people that are working in that codebase for those clean up jobs? I understand (I think) where you are coming from but I am unclear there is a huge benefit here and I have had merge conflict in files I was working on thanks to those cleanups. 
It seems just fair to me to involve the people that are working in those files
(@tnowacki who I added in the review)

File: language/move-lang/src/naming/translate.rs [hunk 1]
```diff
@@ -305,16 +305,18 @@ fn type_parameters(context: &mut Context, type_parameters: Vec<(Name, Kind)>) ->
             let id = N::TParamID::next();
             let debug = name.clone();
             let tp = N::TParam { id, debug, kind };
-            match context.tparams.add(name.clone(), tp.clone()) {
```
Comment (tnowacki) [inline language/move-lang/src/naming/translate.rs]: I'd prefer this to be an `if let`

File: language/move-lang/src/typing/translate.rs [hunk 2]
```diff
@@ -67,10 +67,7 @@ fn main_function(
     main: Option<(Address, FunctionName, N::Function)>,
 ) -> Option<(Address, FunctionName, T::Function)> {
     context.current_module = None;
-    match main {
-        None => None,
-        Some((addr, name, f)) => Some((addr, name.clone(), function(context, name, f))),
-    }
+    main.map(|(addr, name, f)| (addr, name.clone(), function(context, name, f)))
```
Comment (tnowacki) [inline language/move-lang/src/typing/translate.rs]: this one is fine. I like this one

**Comment by huitseeker** [comment]
@dariorussi @tnowacki Sure! Will do!

**Comment by huitseeker** [comment]
@tnowacki ok. Why the request changes?

File: crypto/crypto-derive/src/unions.rs [hunk 3]
```diff
@@ -103,33 +103,31 @@ pub fn impl_enum_validkey(name: &Ident, variants: &DataEnum) -> TokenStream {
...
-    Err(syn::Error::new(
-        proc_macro2::Span::call_site(),
-        format!("Could not find attribute {}", attr_name),
-    ))
+            bad => Err(syn::Error::new_spanned(
+                bad,
+                &format!("Could not parse {} attribute", attr_name)[..],
+            )),
+        })
+        .unwrap_or_else(|| {
```
Comment (zekun000) [inline crypto/crypto-derive/src/unions.rs]: it took me a while to realize this unwraps the `find` not the `map` result, can we use map_or_else instead?

**Comment by tnowacki** [comment]
> @tnowacki ok. Why the request changes?

The request changes was for the first set of changes, nothing after

**Pull request closed without merging**